### DISCUSSION TITLE
Set initial value of `deflisten` variable as result of a command

### DISCUSSION THREAD

**Discussion by METROIDHunter9000** [elkowar/eww#911] [Q&A] [answered]
Hi,

Is there a way to set the initial value of a `deflisten` variable not to a string, but to the result of a given shell command?
With my use case, I'd like my deflisten variable to call a shell script to get an initial value, and for the rest of its life it will listen for changes to a file with `tail -f`.

But as far as I can tell, the only way to execute shell commands and pass results to eww is by using `defpoll` and the listening function of `deflisten`. I would like to avoid using `defpoll`, and `deflisten` doesn't work for an initial value in my case since I'm running `tail -f` on a file in /tmp which won't exist on a fresh reboot.

Is this possible within eww? Is there some trick I could use to achieve this result?

**Comment by w-lfchen** [comment, marked as answer]
I'd recommend just changing your script a little, simply `echo <initial value>` and then the `tail` bit.

**Reply by METROIDHunter9000** [reply]
Brilliant, that works! Makes sense now that you mention it. Thank you
